api/data.go

```diff
@@ -67,3 +67,24 @@ func (api *API) AddDataSets(ctx context.Context) http.HandlerFunc {
 		}
 	}
 }
+
+// GetCacheTime retrieves cache time data for a given ID and writes it to the HTTP response.
+func (api *API) GetCacheTime(ctx context.Context, w http.ResponseWriter, req *http.Request) {
+	log.Info(ctx, "calling get cache time handler")
+
+	vars := mux.Vars(req)
+	id := vars["id"]
+
+	cacheTime, err := api.dataStore.GetCacheTime(ctx, id)
+	if err != nil {
+		log.Error(ctx, "error retrieving cache time from datastore", err)
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		return
+	}
```

Comment:
Think this needs to differentiate if there's been an error or it just doesn't exist. There's performance implications to this, as 500 errors are re-tried by default.